schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java

```diff
@@ -6,12 +6,58 @@
 
 import java.util.Map;
 
+/**
+ * Enrollment Service interface provides methods for enroll and unenroll enrollments, fulfill milestones and for retrieving
+ * alert timings.
+ */
 public interface EnrollmentService {
-    Long enroll(String externalId, String scheduleName, String startingMilestoneName, DateTime referenceDateTime, DateTime enrollmentDateTime, Time preferredAlertTime, Map<String, String> metadata);
 
+    /**
+     * Creates the enrolment from the given parameters and schedule alerts jobs. This method also sends
+     * {@link org.motechproject.scheduletracking.events.EnrolledUserEvent} if there is no active enrolment for the
+     * external id and the schedule name. If the user has already enrolled for the same schedule which is currently
+     * active, then enrollment is updated and alerts are rescheduled.
+     *
+     * @param externalId the client external id
+     * @param scheduleName the name of the schedule
+     * @param startingMilestoneName the starting milestone name
+     * @param referenceDateTime the reference date and time
```

Comment:
The javadoc should shed more light on the significance of these 3 datetime parameters